pyedflib/highlevel.py

```diff
@@ -441,6 +441,10 @@ def write_edf(edf_file, signals, signal_headers, header=None, digital=False,
         'file_type must be in range -1, 3'
     assert block_size<=60 and block_size>=-1 and block_size!=0, \
         'blocksize must be smaller or equal to 60'
+    if file_type != -1:
```

Comment:
Added a check here in write_edf to make sure file extension matches file_type opt.